### DISCUSSION TITLE
[Feature Request] Raise error or Warning when trying to resolve a context name that's not defined

### DISCUSSION THREAD

**Discussion by z64555** [community/community#12352] [Actions]
So, I just spent all of this morning debugging a new workflow for our organization and finally determined why the env were not being set from `workflow_dispatch` inputs. :(

As it turns out, if the context name isn't 100% valid, the workflow silently assumes a null value for the undefined/unrecognized variable.  

For example, `${{ github.events.inputs.releaseTag }}`is not the same as `${{ github.event.inputs.releaseTag }}` and for that matter the former is not defined.

This feature request would thus have undefined context names raise an error or warning whenever the .yaml parser detects them in order to aide debugging efforts.  Since some workflows may depend on this silent behavior, the feature may need to be an opt-in.

Edit: I apologize if this is in the wrong board, I'm looking for the board to provide feedback about the Github Actions feature.

**Comment by ethomson** [comment]
Ouch, I'm sorry to hear about the trouble. 😞  I agree that this is problematic. Improving the errors that our workflow parser produces -- and, in this case, actually _producing_ one -- is something that I'm going to start working on soon. Thanks much for the feedback, and apologies again.

**Comment by link2xt** [comment]
Somewhat related, referenced nonexistent variable here: https://github.com/deltachat/deltachat-core-rust/commit/11e6a325a2e50026d77863c14fdf0ede8fa71fcd

Resulted in artifact with names like `macos-latest--libdeltachat.a`, everything worked but should have errored out instead.
